### DISCUSSION TITLE
Kubernetes 1.25 Major Themes

### DISCUSSION THREAD

**Discussion by csantanapr** [kubernetes/sig-release#1925] [General]
This is a discussion to discuss Major Themes for the **v1.25** release.

Here is the [discussion](https://github.com/kubernetes/sig-release/discussions/1868) from the last release cycle.

**Comment by leonardpahlke** [comment]
cc @kubernetes/release-team-leads

**Comment by cici37** [comment]
CRD Validation Expression Language is promoted to beta in 1.25

**Reply by csantanapr** [reply]
Do you have some links as references for the end user to know more about?

**Reply by cici37** [reply]
The documentation is available here: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules
Thank you

**Comment by saschagrunert** [comment]
SeccompDefault promoted to beta in v1.25: https://deploy-preview-34290--kubernetes-io-main-staging.netlify.app/docs/tutorials/security/seccomp/#enable-the-use-of-runtimedefault-as-the-default-seccomp-profile-for-all-workloads

**Comment by ameukam** [comment]
Moving from k8s.gcr.io to registry.k8s.io got merged in 1.25: https://github.com/kubernetes/kubernetes/pull/109938

More info: https://github.com/kubernetes/k8s.io/wiki/New-Registry-url-for-Kubernetes-(registry.k8s.io)

Announcement: https://groups.google.com/a/kubernetes.io/g/dev/c/DYZYNQ_A6_c/m/oD9_Q8Q9AAAJ

**Comment by rikatz** [comment]
endPort field in Network Policy is now promoted to GA

Network Policy providers that support endPort field now can use it to specify a range of ports to apply a Network Policy.

Previously, each Network Policy could only target a single port.

Please be aware that endPort field MUST BE SUPPORTED by the Network Policy provider. In case your provider does not support endPort and this field is specified in a Network Policy, the Network Policy will be created covering only the port field (single port).

Ref.: https://github.com/kubernetes/kubernetes/pull/110868

**Comment by xing-yang** [comment]
We'd like to include the following in 1.25 major themes from sig-storage:
- The [Local Ephemeral Storage Capacity Isolation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/361-local-ephemeral-storage-isolation) feature moved to GA in 1.25. This was introduced as alpha in 1.8, moved to beta in 1.10, and it is now a stable feature. It provides support for capacity isolation of local ephemeral storage between pods, such as EmptyDir, so that a pod can be hard limited in its consumption of shared resources by evicting Pods if its consumption of local ephemeral storage exceeds that limit.
- [CSI Migration](https://kubernetes.io/blog/2021/12/10/storage-in-tree-to-csi-migration-status-update/#quick-recap-what-is-csi-migration-and-why-migrate) is an on-going effort that SIG Storage has been working on for a few releases now. The goal is to move in-tree volume plugins to out-of-tree CSI drivers and eventually remove the in-tree volume plugins. The [core CSI Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) feature moved to GA in 1.25. CSI Migration for GCE PD and AWS EBS also moved to GA. CSI Migration for vSphere remains in beta but is on by default in 1.25. CSI Migration for Portworx moved to Beta (but is off-by-default) in 1.25.
- [Container Object Storage Interface](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-supportt-storage-interface-spec) (COSI) was introduced as an alpha feature in 1.25. COSI enables provisioning and access to object storage in Kubernetes alongside file and block storage. Similar to CSI, it introduces gRPC interfaces for object storage providers to write drivers to provision object storage.

cc @saad-ali @msau42 @jsafrane

**Comment by xing-yang** [comment]
We'd like to add one more GA feature from sig-storage:
- The [CSI Ephemeral Volume](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/596-csi-inline-volumes) feature allows CSI volumes to be specified directly in the pod specification for ephemeral use cases. They can be used to inject arbitrary states, such as configuration, secrets, identity, variables or similar information, directly inside pods using a mounted volume.  This was initially introduced in 1.15 as an alpha feature, and it moved to GA in 1.25. This feature is used by some CSI drivers such as the [secret-store CSI driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver).

cc @saad-ali @msau42 @jsafrane

**Comment by marosset** [comment]
During The 1.25 release SIG-Windows has made a lot of improvements around testing including:
- http://perf-dash.k8s.io/#/?jobname=soak-tests-capz-windows-2019 now reports performance data for Windows!
- A lot of progress has been made towards running Kubernetes unit tests on Windows machines during CI (tracked with https://github.com/kubernetes/kubernetes/issues/51540)
- More conformance tests can now run on Windows nodes (https://github.com/kubernetes/kubernetes/pull/108592)
- New repository created for Windows Operational Readiness work (https://github.com/kubernetes-sigs/windows-operational-readiness)

cc @jsturtevant @jayunit100 @claudiubelu

**Comment by ritazh** [comment]
1.25 major themes from sig-auth:
- The PodSecurity admission plugin has graduated to GA and is enabled by default. The admission configuration version has been promoted to `pod-security.admission.config.k8s.io/v1`.
  - https://github.com/kubernetes/kubernetes/pull/110459
  - See https://kubernetes.io/docs/concepts/security/pod-security-admission/ for usage guidelines.
- The beta `PodSecurityPolicy` admission plugin, deprecated since 1.21, is removed. Follow the instructions at https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/ to migrate to the built-in PodSecurity admission plugin (or to another third-party policy webhook) prior to upgrading to v1.25.
  - https://github.com/kubernetes/kubernetes/pull/109798
- As of v1.25, the PodSecurity `restricted` level no longer requires pods that set .spec.os.name="windows" to also set Linux-specific securityContext fields. If a 1.25+ cluster has unsupported [out-of-skew](https://kubernetes.io/releases/version-skew-policy/#kubelet) nodes prior to v1.23 and wants to ensure namespaces enforcing the `restricted` policy continue to require Linux-specific securityContext fields on all pods, ensure a version of the `restricted` prior to v1.25 is selected by labeling the namespace (for example, `pod-security.kubernetes.io/enforce-version: v1.24`)
  - https://github.com/kubernetes/kubernetes/pull/105919
- Adds KMS v2alpha1 support
  - https://github.com/kubernetes/kubernetes/pull/111126
  - https://github.com/kubernetes/website/pull/35046
- Encrypt data with DEK using AES-GCM instead of AES-CBC for kms data encryption. No user action required. Reads with AES-GCM and AES-CBC will continue to be allowed. 
  - https://github.com/kubernetes/kubernetes/pull/111119
  - https://github.com/kubernetes/website/pull/35046
- The `gcp` and `azure` auth plugins have been removed from client-go and kubectl. See https://github.com/Azure/kubelogin and https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke for details about the cloud-specific replacements.
  - https://github.com/kubernetes/kubernetes/pull/110013

cc @tallclair @liggitt @enj @mikedanese @deads2k

**Reply by csantanapr** [reply]
Hi @ritazh are these all the changes from sig-auth for v1.25 or only major themes?

**Reply by ritazh** [reply]
Not all the changes, only major themes.

**Reply by csantanapr** [reply]
Thank you @ritazh for confirming
